rs/registry/canister/src/request_impls.rs

```diff
@@ -0,0 +1,8 @@
+use crate::pb::v1::{GetSubnetForCanisterRequest, SubnetForCanister};
+use ic_nervous_system_clients::Request;
+
+impl Request for GetSubnetForCanisterRequest {
```

Comment:
To avoid creating circular dependencies if we want to use the agent or request libraries in the tests for these canisters, we might just move all the impl Request into the ic_nervous_system_clients crate?

Reply:
Agreed. Since this isn't the first time `impl Request` is added to canister crates, I'm going to 
1. Merge this as-is.
2. Make a PR to move all these impls out of the canister crates and into ic_nervous_system_clients